server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateIntervalWrapper.java

```diff
@@ -69,27 +75,51 @@ public void writeTo(StreamOutput out) throws IOException {
         public String value() {
             return name().toLowerCase(Locale.ROOT);
         }
+
+        public boolean isValid() {
+            // I'm being a little cheeky here and just reusing the name for signaling invlaid choices too
+            return this.preferredName != null;
+        }
+
+        public String getPreferredName() {
+            if (preferredName == null) {
+                throw new IllegalStateException("Invalid use of legacy date histogram interval");
+            }
+            return preferredName;
+        }
+
+        private String preferredName;
+        IntervalTypeEnum(String preferredName) {
+            this.preferredName = preferredName;
+        }
     }
 
     private DateHistogramInterval dateHistogramInterval;
     private IntervalTypeEnum intervalType = IntervalTypeEnum.NONE;
 
     public static <T extends DateIntervalConsumer> void declareIntervalFields(ObjectParser<T, String> parser) {
-
-        // NOTE: this field is deprecated and will be removed
+        /*
+         REST version compatibility.  When in V_7 compatibility mode, continue to parse the old style interval parameter,
+         but immediately adapt it into either fixed or calendar interval.
+         */
         parser.declareField((wrapper, interval) -> {
+            DEPRECATION_LOGGER.deprecate(DeprecationCategory.AGGREGATIONS, "date-interval-getter", DEPRECATION_TEXT);
             if (interval instanceof Long) {
-                wrapper.interval((long) interval);
+                wrapper.fixedInterval(new DateHistogramInterval(interval + "ms"));
             } else {
-                wrapper.dateHistogramInterval((DateHistogramInterval) interval);
+                try {
```

Comment:
I wonder if we need try catch logic. Maybe we could extract https://github.com/elastic/elasticsearch/pull/75000/files#diff-dd9ebdfdc17eaf7e26fe7cb69a5695245895ff84a1b268b75da2937628f7d8e7R165 check and choose if calendarInterval or fixedInterval should be called?

Reply:
That link isn't working, but I'm pretty sure I know what you mean.  I replaced the try/catch with an if/else.